File: results/dnd-rest-benchmark/lifecycle-runs/20260804T081334Z_claude_claude-sonnet-5_php-symfony/src/Storage/StorageController.php
<?php

namespace App\Storage;

use Symfony\Component\HttpFoundation\JsonResponse;

/** Handlers for the storage introspection/reset endpoints (/v1/storage/*). */
final class StorageController
{
    public function status(): JsonResponse
    {
        $db = Database::connection();
        $stmt = $db->prepare('SELECT value FROM schema_meta WHERE key = ?');
        $stmt->execute(['initialized']);
        $initialized = $stmt->fetchColumn();

        return new JsonResponse([
            'driver' => 'sqlite',
            'schema_version' => Database::SCHEMA_VERSION,
            'initialized' => $initialized === '1',
        ]);
    }

    public function reset(): JsonResponse
    {
        Database::resetSchema(Database::connection());

        return new JsonResponse([
            'ok' => true,
            'schema_version' => Database::SCHEMA_VERSION,
        ]);
    }
}

File: results/dnd-rest-benchmark/lifecycle-runs/20260731T051117Z_claude_claude-sonnet-5_ruby-rails/app/controllers/storage_controller.rb
# Introspection and reset for the SQLite-backed game state.
class StorageController < ApplicationController
  def storage_status
    render json: {
      driver: 'sqlite',
      schema_version: SCHEMA_VERSION,
      initialized: GameStorage.initialized ? true : false
    }
  end

  def storage_reset
    GameStorage.reset
    render json: { ok: true, schema_version: SCHEMA_VERSION }
  end
end
